lib/connection.js

```diff
@@ -647,8 +662,9 @@ Connection.prototype.readBytes = function(buffer, length) {
 
 Connection.prototype.parseCString = function(buffer) {
   var start = this.offset;
-  while(buffer[this.offset++] !== 0) { }
-  return buffer.toString(this.encoding, start, this.offset - 1);
+  var end = indexOf(buffer, 0, start);
+  this.offset = end + 1;
```

Comment:
I didn’t want to throw since none of the other parsers do yet… it’s a bit sticky, to be sure.